lib/autobuild/import/git.rb

```diff
@@ -293,6 +293,52 @@ def validate_importdir(package)
             git_dir(package, true)
         end
 
+        
+        # Return default local branch if exists, if not return nil
+        #
+        # @param [Package] package
+        def default_local_branch(package)
+            ls_local_string = run_git(package, 'symbolic-ref', "refs/remotes/#{@remote_name}/HEAD").first.strip
+            default_local_branch = ls_local_string.match("refs/remotes/#{@remote_name}/(.*)")
+            default_local_branch.nil? ? nil : default_local_branch[1]
+            rescue Autobuild::SubcommandFailed
+                return nil
+        end
+
+        # Return default remote branch if exists, if not return 'master'
+        #
+        # @param [Package] package
+        def default_remote_branch(package)
+            ls_remote_string = package.run(:import,
+                Autobuild.tool('git'), 'ls-remote', '--symref', repository).first.strip
+            default_branch_match = ls_remote_string.match("ref:[^A-z]refs/heads/(.*)[^A-z]HEAD")
+            if !default_branch_match.nil?
+                return default_branch_match[1]
+            else
+                return 'master'
+            end
+        end
+
+        # Return default local branch if exists, if not return the default remote branch
+        #
+        # @param [Package] package
+        def default_branch(package)
+            local_branch = default_local_branch(package)
+            if local_branch.nil?
+                return default_remote_branch(package)
+            else
+                return local_branch
+            end
+        end
+
+        def has_all_branches?
```

Comment:
This is just

```ruby
def has_all_branches?
    remote_branch && local_branch
end
```

Reply:
A curiosity, should a boolean method return `nil` instead of false in ruby? I am asking this because `nil && nil => nil`.

Reply:
In Ruby `nil` and `false` are the only things that evaluates to falsy (even `0` is truthy). So, in boolean contexts they mean the same.